main.go

```diff
@@ -83,64 +75,42 @@ func currentNDKHome() string {
 	return "ndk-bundle"
 }
 
-func newNDKHome() (string, error) {
-	if v := os.Getenv("ANDROID_HOME"); v != "" {
-		// $ANDROID_HOME is deprecated
-		return filepath.Join(v, "ndk-bundle"), nil
-	}
-	if v := os.Getenv("ANDROID_SDK_ROOT"); v != "" {
-		// $ANDROID_SDK_ROOT is preferred over $ANDROID_HOME
-		return filepath.Join(v, "ndk-bundle"), nil
-	}
-	return "", errors.New("neither $ANDROID_HOME nor $ANDROID_SDK_ROOT is specified")
-}
-
-// updateNDK downloads and extracts the required NDK revision (if not already installed to the correct location).
-// NDK is installed to the `ndk-bundle` subdirectory of the SDK location, while updating $ANDROID_NDK_HOME for
+// updateNDK installs the requested NDK version (if not already installed to the correct location).
+// NDK is installed to the `ndk/version` subdirectory of the SDK location, while updating $ANDROID_NDK_HOME for
 // compatibility with older Android Gradle Plugin versions.
 // Details: https://github.com/android/ndk-samples/wiki/Configure-NDK-Path
-func updateNDK(revision string) error {
-	ndkDownloadURL := ndkDownloadURL(revision)
+func updateNDK(version string, androidSdk *sdk.Model) error {
 	currentNdkHome := currentNDKHome()
 
-	currentRevision := ndkRevision(currentNdkHome)
-	if currentRevision == revision {
-		logger.Donef("NDK r%s already installed at %s", revision, currentNdkHome)
+	currentVersion := ndkVersion(currentNdkHome)
+	if currentVersion == version {
+		logger.Donef("NDK %s already installed at %s", version, currentNdkHome)
 		return nil
 	}
 
-	if currentRevision != "" {
-		logger.Printf("NDK version %s found at: %s", currentRevision, currentNdkHome)
+	if currentVersion != "" {
+		logger.Printf("NDK %s found at: %s", currentVersion, currentNdkHome)
 	}
 
-	logger.Printf("Removing NDK...")
+	logger.Printf("Removing existing NDK...")
```

Comment:
Do we still need to remove old NDK in this brave new world?

Reply:
I think it's a good idea, because the preinstalled NDK is at `ndk-bundle`, so without removing it there would be 2 NDK folders. Not sure what would get the priority.

Reply:
Are there Step changes needed when we update the Stacks to this new format? (ndk instead of ndk-bundle).
Should the Step support the new layout out of the box?

Reply:
Discussed offline: As long ANDROID_NDK_HOME is set, the Step will find it. To support multiple preinstalled NDK versions is not yet supported.